### ISSUE TITLE
Solve with a source term from file

### ISSUE THREAD

**Issue by giovastabile** [NaluCFD/Nalu#332]
Is it possible to solve a problem with a source term coming from the output file of another simulation? I add a look to the code but I couldn't understand whether this is possible or not.

**Comment by spdomin** [comment]
I believe that if you have the field existing in an exodus data base, you can use the input/output feature to extract this data to a field, see:

https://github.com/NaluCFD/Nalu/tree/master/reg_tests/test_files/elemBackStepLRSST 

or the last help ticket, https://github.com/NaluCFD/Nalu/issues/311

If you then declare the field in one of the equation systems, you can transfer into that field. Finally, you would need to code a source term.

@jamelvin may have coded a general source term, however, that may be only through a MASA interface.

**Comment by giovastabile** [comment]
Thanks for the prompt response.

What I want to do is to solve a Poisson problem with a source term generated from a previous simulation and stored in an exodus data base. 

If I understood it well I would need to create an input file with two realms, one pointing to my self made system of equations (the one with the additional source term) and one which is of type input/output...

Am I correct?

**Comment by spdomin** [comment]
Yes, I believe so. You can probably register the field in the LowMachEQS.

**Comment by giovastabile** [comment]
I tried to implement your suggestions but I'm stuck and maybe I miss something. Unfortunately I'm using an old version of Nalu for which is not present the input/output feature. I'm modifying an existing system of equations in order to read a source term from an existing exodus file. I came up with the solution reported in the code but when I try to print out the read field I have only 0 even though the field is not. Do you have any idea concerning what is missing? Best and thanks for the help, Giovanni 
                
std::string fileName = "prova.e";
std::string fieldName = "poissonPot";
int interpolation_intervals = 1;

stk::ParallelMachine pm = NaluEnv::self().parallel_comm();
stk::io::StkMeshIoBroker mesh_data(pm);
size_t resultsFileIndex = mesh_data.add_mesh_database(fileName, stk::io::READ_MESH);
mesh_data.set_active_mesh(resultsFileIndex);
mesh_data.create_input_mesh();
stk::mesh::MetaData &meta_data = mesh_data.meta_data();
const int numberOfStates = 1;
stk::mesh::Field<double> &field0 = meta_data.declare_field<stk::mesh::Field<double> >(stk::topology::NODE_RANK, fieldName, numberOfStates);
stk::mesh::put_field(field0, meta_data.universal_part());

mesh_data.populate_bulk_data();
size_t results_output_index = mesh_data.create_output_mesh("ciao.e", stk::io::WRITE_RESULTS);

typedef stk::mesh::Field<double>  ScalarFieldType;
ScalarFieldType* field1 = meta_data.get_field<ScalarFieldType>(stk::topology::NODE_RANK, fieldName);

stk::mesh::BulkData* bulk_data = &mesh_data.bulk_data();
stk::mesh::Selector s_all_nodes = meta_data.locally_owned_part();
stk::mesh::BucketVector node_buckets = bulk_data->get_buckets( stk::topology::NODE_RANK, s_all_nodes );

std::vector<stk::mesh::Entity> solid_nodes;

for (stk::mesh::BucketVector::const_iterator ib = node_buckets.begin(); ib != node_buckets.end(); ++ib)
{
 stk::mesh::Bucket & b = **ib;
 const stk::mesh::Bucket::size_type length = b.size();
  for (unsigned int i = 0; i < b.size(); ++i)
  {
     solid_nodes.push_back(b[i]);
  }
 }

for (unsigned int i = 0; i < solid_nodes.size(); i++)
{
 double* prop = (double*) stk::mesh::field_data(*field1, solid_nodes[i]);
 std::cout << solid_nodes[i] << std::endl;
 std::cout << *prop << std::endl;
}

**Comment by sayerhs** [comment]
@giovastabile Can you try adding the following line to the `StkMeshIoBroker` instance to tell it to load fields from the file and try again?

```
stk::ParallelMachine pm = NaluEnv::self().parallel_comm();
stk::io::StkMeshIoBroker mesh_data(pm);
size_t resultsFileIndex = mesh_data.add_mesh_database(fileName, stk::io::READ_MESH);
mesh_data.set_active_mesh(resultsFileIndex);
mesh_data.create_input_mesh();

/** Load all fields present in the mesh database **/
mesh_data.add_all_mesh_fields_as_input_fields(); 
```

**Comment by giovastabile** [comment]
I tried but it I still have the same issue. I also attach the exodus file
from which I'm trying to extract the field.

Il 04 mar 2018 6:00 PM, "Shreyas Ananthan" <notifications@github.com> ha
scritto:

**Comment by spdomin** [comment]
You need to update to a new version. The input/output is available either in the master or v1.1.0-release. Once you have a version that supports the capability, we can walk through the use case.

Also, please do note the project/use-case driving the capability as this would help understand how Nalu is being used.
